util/git/client.go

```diff
@@ -361,9 +395,8 @@ func (m *nativeGitClient) runCredentialedCmd(command string, args ...string) (st
 func (m *nativeGitClient) runCmdOutput(cmd *exec.Cmd) (string, error) {
 	cmd.Dir = m.root
 	cmd.Env = append(cmd.Env, os.Environ()...)
-	cmd.Env = append(cmd.Env, "HOME=/dev/null")
-	cmd.Env = append(cmd.Env, "GIT_CONFIG_NOSYSTEM=true")
-	cmd.Env = append(cmd.Env, "GIT_CONFIG_NOGLOBAL=true")
```

Comment:
I have removed these environment variables, since I was assuming these were set to not fiddle with Git configuration when the `argocd` CLI still tested repository access from the client it was executed. This behaviour was removed with PR #1807, and the `git`  is not called on the client anymore.

Reply:
I think this is right assumption and we don't need these variables anymore. @jessesuen can you confirm please?